uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/leave-calendar.html

```diff
@@ -88,6 +88,7 @@
       <div ng-if="calendar.injectMonths">
         <leave-calendar-month ng-repeat="month in calendar.months track by $index"
           contacts="calendar.contacts"
+          contact-ids-to-reduce-to="calendar.contactIdsToReduceTo"
```

Comment:
`contacts` attribute should pass the filtered contacts. This should not be a responsibility of `leave-calendar-month`. `contact-ids-to-reduce-to` should be removed and the filtering logic should be moved to parent controller.

Reply:
Unfortunately, leave-calendar-month loads Leave Requests. The new logic for additional filtering of Contacts is *Contacts have either Leave Requests OR Contracts*. Leave requests are loaded in the `leave-calendar-month` while Contracts **have to be loaded** in `admin-leave-calendar` since this logic relates to the Admin only. We had this discussion with @AkA84 and it seems that this is the only proper way to go.

Reply:
After having a call with @igorpavlov this is the best solution we can provide so far without blocking the ticket for too long, and this feature is critical. But there needs to be refactoring done because the logic to display contacts is so confusing that it needs to be explained in a call, it can't be inferred by simply looking at the components.

Here are some of my suggestions for a future ticket:

* Stop using controllers as services:
```js
   // leave-calendar.component.js #131
    function injectSubController () {
      subController = $controller('LeaveCalendar' + _.capitalize(userRole) + 'Controller').init(vm);
    }
```

Instead of using controllers for each role, these should be services (ex: `adminLeaveCalendarContactsService`). 

* A map **value** can be created to retrieve the proper service. This:
```js
// note: the services names are examples
module.value('leaveCalendarContactsByRole', function (adminService, managerService, staffService) {
  return { admin: adminService, manager: managerService, staff: staffService });
});

//and use it like so:
contactService = leaveCalendarContactsByRole['admin'];
```

Instead of:
```js
subController = $controller('LeaveCalendar' + _.capitalize(userRole) + 'Controller').init(vm);
```

* the services should return the contacts that are to be displayed on the calendar. The component should not reduce any value or apply any filters to the contacts to display. This will make the component unaware of any business logic of the contacts to display depending on the role, and just focuses on displaying the contacts it was passed as an attribute or retrieved from a service.